software/firmware/configuration.py

```diff
@@ -14,8 +14,14 @@
 from collections import namedtuple
 
 Validation = namedtuple("Validation", "is_valid message")
```

Comment:
[discussion] Shouldn't the docstring go above the definition?

Reply:
Yeah, I had them that way at first, but it doesn't work. This is how the doc generator expects them. I believe it is to mirror the location of the docstring below a class or function declaration.

Reply:
Ah, perhaps it's interpreting it as a sort of function docstring instead of a sort of const/var docstring. Either way, if it works, then LGTM.